src/main/java/com/powsybl/openloadflow/sa/WoodburyDcSecurityAnalysis.java

```diff
@@ -0,0 +1,298 @@
+                                                                                          Map<String, ComputedContingencyElement> contingencyElementByBranch, DenseMatrix flowStates, DenseMatrix contingenciesStates,
+                                                                                          ReportNode reportNode) {
+
+        PropagatedContingency contingency = connectivityAnalysisResult.getPropagatedContingency();
+        Set<LfBus> disabledBuses = connectivityAnalysisResult.getDisabledBuses();
+
+        // as we are processing a contingency with connectivity break, we have to reset active power flow of a hvdc line
+        // if one bus of the line is lost.
+        for (LfHvdc hvdc : loadFlowContext.getNetwork().getHvdcs()) {
+            if (Networks.isIsolatedBusForHvdc(hvdc.getBus1(), disabledBuses) ^ Networks.isIsolatedBusForHvdc(hvdc.getBus2(), disabledBuses)) {
+                contingency.getGeneratorIdsToLose().add(hvdc.getConverterStation1().getId());
+                contingency.getGeneratorIdsToLose().add(hvdc.getConverterStation2().getId());
+            }
+        }
+
+        return calculatePostContingencyStatesForAContingency(loadFlowContext, contingenciesStates, flowStates,
+                contingency, contingencyElementByBranch, disabledBuses, connectivityAnalysisResult.getElementsToReconnect(), reportNode,
+                connectivityAnalysisResult.getPartialDisabledBranches());
+    }
+
+    private double[] getStatesAsArray(DenseMatrix states) {
+        double[] angleStatesArray = new double[states.getRowCount()];
+        for (int i = 0; i < angleStatesArray.length; i++) {
+            angleStatesArray[i] = states.get(i, 0);
+        }
+        return angleStatesArray;
+    }
+
+    /**
+     * Filter the contingencies applied on the given network.
+     * The contingencies on buses are not supported in the {@link ConnectivityBreakAnalysis}, so we must add
+     * the linked branches in the contingency elements.
+     * Contingencies on switch are not yet supported in {@link WoodburyDcSecurityAnalysis}.
+     */
+    private void filterPropagatedContingencies(LfNetwork lfNetwork, List<PropagatedContingency> propagatedContingencies) {
+
+        // contingencies on switch not yet supported
+        propagatedContingencies.stream()
+                .flatMap(contingency -> contingency.getBranchIdsToOpen().keySet().stream())
+                .map(branchId -> lfNetwork.getBranchById(branchId).getBranchType())
+                .filter(branchType -> branchType == LfBranch.BranchType.SWITCH)
+                .findAny()
+                .ifPresent(e -> {
+                    throw new IllegalArgumentException("Contingencies on switch not yet supported in fast DC Security Analysis");
+                });
+
+        // map contingencies on buses to contingencies on linked branches terminals
+        propagatedContingencies.stream()
```

Comment:
I think this is not what you have to do. You have to used `cleanContingencies` from DcSensitivityAnalysis. Right? If yes, I suggest to put it as utility method in PropagatedContingency.

Reply:
Work in progress in PR #1081.